processor/memorylimiter/memorylimiter.go

```diff
@@ -238,44 +239,89 @@ func (ml *memoryLimiter) forcingDrop() bool {
 	return atomic.LoadInt64(&ml.forceDrop) != 0
 }
 
-func (ml *memoryLimiter) memCheck() {
-	ms := ml.readMemStats()
-	ml.memLimiting(ms)
+func (ml *memoryLimiter) setForcingDrop(b bool) {
+	var i int64
```

Comment:
Just as an idea: if you use CompareAndSwap, you can simplify your `checkLimits` func (see comment there). Also, why `int64` and not perhaps `int32`?

Reply:
No need for CAS, just a `SwapInt64` is enough for your idea.

Reply:
Is `SwapInt64` a noop if the new value equals the old value?

Reply:
Reworked a bit. I don't want to use CAS or dawp, it makes the logic difficult to understand, especially since there is no bool version.